server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/operationhandler/EcoreLabelEditOperationHandler.java

```diff
@@ -61,9 +67,15 @@ public void executeOperation(ApplyLabelEditOperation editLabelOperation, Graphic
 								"No shape element for label with id " + editLabelOperation.getLabelId() + " found");
 		
 						if (node_semantic instanceof EClassifier) {
+							String oldRootURI = EcoreUtil.getURI(node_semantic).fragment().toString().substring(2);
```

Comment:
Avoid substrings in the middle of the code. If this is required, this should be moved to a separate (util) method that does only that.